### DISCUSSION TITLE
there is no interactive perject-close / perject-close-collection?

### DISCUSSION THREAD

**Discussion by yuuyins** [overideal/perject#1] [Q&A] [answered]
82d007e

i do not use transient. there used to be an interactive perject-close-collection

**Comment by yuuyins** [comment]
well just loaded perject-transient.

perject-close: `there currently is no collection`, even though i have a collection opened.

**Reply by overideal** [reply]
That could be a bug. The collections and their projects are saved in the variable `perject-collections`, which is updated whenever appropriate (e.g. when opening or closing a collection). What probably happened in your case is that the variable got altered sometime before but the actual "state" (e.g. which collections are open) was not changed like expected. In this sense, if there is a bug, it must have happened before, `perject-close` not working is only the symptom.
Maybe you can try to inspect the problem using this variable and give me a way to reproduce the issue. Then I can surely fix it :)

**Reply by yuuyins** [reply]
maybe incompatible changes between the revisions? perject creates the .lock file, but does not add to `perject-collections`. this is for collections created with the older revision i was in. newer collections created with current commit work alright.

i think this will not be easy reproduce, since it requires my personal perject collections which have many content. i will see

**Comment by overideal** [comment, marked as answer]
Hi, yes there was an interactive perject-close-collection (now it is not interactive anymore). https://github.com/overideal/perject/commit/366b2c129ea0c04fd4239a3de45bf5eb489629b9 introduces the interactive command perject-close as an enhanced replacement. The reason to use transient is to give the user more options (e.g. whether to kill the buffers associated to the collection). I understand that not everyone wants to use transient (though I think it is superb). Put this function into your init file and you get essentially the old function back:
```
(defun perject-close-quick (name)
  "Close the collection named NAME.
Use `perject-close-default' to alter the behavior of this command."
  (interactive
   (list
    (perject--get-collection-name
     "Close collection: " 'active nil t (car (perject-current))
     "There currently is no collection to close"
	 "No collection specified")))
  (apply #'perject-close-collection name perject-close-default))
```
You would need to define the analogous functions for deleting and reloading a project if you never wanna use transient.
